### PR TITLE
[RG-223] Add library and compile unit for simulation files

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Add library and compile unit for simulation files.
This changes related to new project wizard and project setting.

 ### Which part of the code base require a change
 
 - [x] Library: newproject
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [x] Regression tests
 - [ ] Continous Integration (CI) scripts

### Screenshots
image

### CODE REVIEW

File: src/NewProject/ProjectManager/project_manager.cpp [hunk 1]
```diff
@@ -1716,7 +1716,100 @@ int ProjectManager::CreateAndAddFile(const QString& suffix,
...
-  const QList<filedata> listFile = opt.sourceFileData.fileData;
+  auto addDesignFiles = [this](const QString& commands, const QString& libs,
+                               const QString& fileNames, int lang,
+                               const QString& grName, bool isFileCopy,
+                               bool localToProject) {
+    setDesignFiles(commands, libs, fileNames, lang, grName, isFileCopy,
+                   localToProject);
+  };
+
+  AddFiles(opt.sourceFileData, addDesignFiles);
```
Comment (ravic-rs) [inline src/NewProject/ProjectManager/project_manager.cpp:1727]: What went into upstream testing with this refactoring? It's not available in the PR description.

**Review by ravic-rs — CHANGES_REQUESTED** [review]
One comment to address.

File: src/NewProject/ProjectManager/project_manager.cpp (hunk 1 above)
Comment (KochynVolodymyr) [reply src/NewProject/ProjectManager/project_manager.cpp:1727]: this change covered by unit testing. And there are actually not much changes. Just some code was moved to separate function.

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**